Doc updates for 0.59 final (#9402)
numba/numba: docs/source/user/troubleshoot.rst

        tmp = [np.complex64(x) for x in range(0)]
        return (tmp, x) # the type of `tmp` is known, but it is still empty

`Object mode` or ``@jit(forceobj=True)`` is too slow
====================================================

> :term:`Object mode` provides little to no speedup compared to regular Python 
> interpretation, and its main point is to allow an internal optimization known as
> :term:`loop-lifting`: this optimization will allow to compile inner
> loops in :term:`nopython mode` regardless of what code surrounds those
> inner loops. The inner loops can still fallback to :term:`object mode` if it 
> uses types or operations that :term:`nopython mode` does not support.

Reviewer: ```suggestion
:term:`object mode` provides little to no speedup compared to regular Python 
interpretation, its main point is to allow an internal optimization known as
:term:`loop-lifting`: . This optimization will allow compilation of inner
loops in :term:`nopython mode` regardless of what code surrounds those
inner loops. The compilation of inner loops can still fallback to
:term:`object mode` if they use types or operations that
:term:`nopython mode` does not support.
```
